src/thread_id.rs

```diff
@@ -13,11 +8,9 @@ unsafe impl lock_api::GetThreadId for RawThreadId {
     const INIT: Self = Self;
 
     fn nonzero_thread_id(&self) -> NonZeroUsize {
-        let thread_id = thread::current().id();
-        assert!(size_of::<ThreadId>() >= size_of::<NonZeroUsize>());
-
-        // TODO: https://github.com/rust-lang/rust/issues/67939
-        let id_ptr = &thread_id as *const ThreadId;
-        unsafe { ptr::read_unaligned(id_ptr as *const NonZeroUsize) }
+        // The address of a thread-local is guaranteed to
+        // be unique to the current thread and non-zero (null)
+        thread_local!(static ID: bool = false);
```

Comment:
We do have to hope that OSe don't reuse this (they do). But I'm not sure it causes UB if so.

Reply:
Don't think so, just need to make sure that no two active threads have the same IDs (as that would break the re-entrant locks).

Reply:
Right. I think that's right too.